plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDistance.java

```diff
@@ -2,55 +2,61 @@
 
 import com.denizenscript.denizen.objects.MaterialTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.type.Leaves;
 import org.bukkit.block.data.type.Scaffolding;
 
-public class MaterialDistance implements Property {
-
-    public static boolean describes(Object material) {
-        return material instanceof MaterialTag
-                && ((MaterialTag) material).hasModernData()
-                && (((MaterialTag) material).getModernData() instanceof Scaffolding
-                || ((MaterialTag) material).getModernData() instanceof Leaves);
+public class MaterialDistance extends MaterialProperty<ElementTag> {
+
+    // <--[property]
+    // @object MaterialTag
+    // @name distance
+    // @input ElementTag(Number)
+    // @description
+    // Controls the horizontal distance between a scaffolding block and the nearest scaffolding block placed above a 'bottom' scaffold,
+    // or between a leaves block and the nearest log (a distance of 7 will cause a leaf to decay if 'persistent' is also false, less than 7 will prevent decay).
+    // -->
+
+    public static boolean describes(MaterialTag material) {
+        BlockData data = material.getModernData();
+        return data instanceof Scaffolding
+                || data instanceof Leaves;
     }
 
-    public static MaterialDistance getFrom(ObjectTag _material) {
-        if (!describes(_material)) {
-            return null;
-        }
-        else {
-            return new MaterialDistance((MaterialTag) _material);
-        }
-    }
+    MaterialTag material;

```

Comment:
Should remove the old field and use the built-in `object` field.

Reply:
And with that, should use the `MaterialProperty` methods (e.g. `getBlockData()` instead of `material.getModernData()`)